src/main/java/org/swisspush/redisques/util/DequeueStatisticCollector.java

```diff
@@ -0,0 +1,100 @@
+package org.swisspush.redisques.util;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.shareddata.AsyncMap;
+import io.vertx.core.shareddata.SharedData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class DequeueStatisticCollector {
+    private static final Logger log = LoggerFactory.getLogger(DequeueStatisticCollector.class);
+    private final static String DEQUEUE_STATISTIC_DATA = "dequeueStatisticData";
+    private final static String DEQUEUE_STATISTIC_LOCK_PREFIX = "dequeueStatisticLock.";
+    private final SharedData sharedData;
+
+    public DequeueStatisticCollector(Vertx vertx) {
+        this.sharedData = vertx.sharedData();
+    }
+
+    public void setDequeueStatistic(final String queueName, final DequeueStatistic dequeueStatistic) {
+        sharedData.getLock(DEQUEUE_STATISTIC_LOCK_PREFIX.concat(queueName)).onSuccess(lock ->
```

Comment:
I think would be cool if we make sure to NOT conceal errors.

I think the main issue is the inlining of the lambdas. Using this code style it is extremely risky to miss error cases (*same counts for using onSuccess()*). If we would use blocks for lambdas and/or "*onComnplete*" instead, those issues would be visible much better. This is because EVERY Promise CAN FAIL. And if it does, we MUST handle (*at very least wrap-and-rethrow the contained exception). If we do not do this, those exceptions will just vanish to nirvana, and we will get just some misterious nullpointers instead without any useful error details.

I give one concrete example here. BUT THERE ARE SEVERAL MORE of this kind. Especially this function here contains A LOT of them.

```patch
--- a/src/main/java/org/swisspush/redisques/util/DequeueStatisticCollector.java
+++ b/src/main/java/org/swisspush/redisques/util/DequeueStatisticCollector.java
@@ -23,8 +21,10 @@ public class DequeueStatisticCollector {
     }

     public void setDequeueStatistic(final String queueName, final DequeueStatistic dequeueStatistic) {
-        sharedData.getLock(DEQUEUE_STATISTIC_LOCK_PREFIX.concat(queueName)).onSuccess(lock ->
-                sharedData.getAsyncMap(DEQUEUE_STATISTIC_DATA, (Handler<AsyncResult<AsyncMap<String, DequeueStatistic>>>) asyncResult -> {
+        sharedData.getLock(DEQUEUE_STATISTIC_LOCK_PREFIX.concat(queueName)).onComplete(lockEv -> {
+            if (lockEv.failed()) throw new RuntimeException("TODO error handling", lockEv.cause());
+            var lock = lockEv.result();
+            sharedData.<String, DequeueStatistic>getAsyncMap(DEQUEUE_STATISTIC_DATA, asyncResult -> {
                 if (asyncResult.failed()) {
                     log.error("Failed to get shared dequeue statistic data map.", asyncResult.cause());
                     lock.release();
```